packages/contracts/development/contracts/RealityETH_ERC20-4.0.sol

```diff
@@ -746,6 +749,41 @@ contract RealityETH_ERC20_v4_0 is BalanceHolder_ERC20, IRealityETHCore_ERC20 {
         withdraw();
     }
 
+    /// @notice Returns true if the supplied history is valid
+    /// @dev Caller must provide the answer history, in reverse order back to the item they want to check
+    /// @dev Not necessarily the entire history
+    /// @dev Useful for freezing an action once a bond is paid for a particular answer, without waiting for resolution
+    /// @dev Cannot be used after the question is finalized
+    /// @param question_id The ID of the question
+    /// @param history_hashes Second-last-to-first, the hash of each history entry. (Final one should be empty).
+    /// @param addrs Last-to-first, the address of each answerer
+    /// @param bonds Last-to-first, the bond supplied with each answer
+    /// @param answers Last-to-first, each answer supplied
+    function isHistoryOfUnfinalizedQuestionValid(
+        bytes32 question_id,
+        bytes32[] memory history_hashes,
+        address[] memory addrs,
+        uint256[] memory bonds,
+        bytes32[] memory answers
+    ) external view stateOpenOrPendingArbitration(question_id) returns (bool) {
```

Comment:
Why wouldn't we allow to prove a history after arbitration?

Reply:
When you do a claim, this erases the history so proving it won't work any more. In theory we could allow you to do this until that happens, but it's a bit of a footgun for developers since people calling reality.eth shouldn't normally have to think about claims, they should only have to think about the question they want to ask and whether it's finalized or not. So I think it's safer to advertise the function as "this is only for questions that aren't finalized" and make it always fail if you test it under finalized conditions.

Once it's finalized you shouldn't need the history any more (ie we're using it because we want to freeze something so it doesn't do any damage while we wait for the result) so this shouldn't cause any harm.